esmvalcore/_recipe.py

```diff
@@ -712,6 +728,31 @@ def _update_extract_shape(settings, config_user):
         check.extract_shape(settings['extract_shape'])
 
 
+def _update_timerange(variable, config_user):
+    if 'timerange' not in variable:
+        return
+
+    timerange = variable.get('timerange')
+    check.valid_time_selection(timerange)
+
+    if '*' in timerange:
+        (files, _, _) = _find_input_files(variable, config_user['rootpath'],
+                                          config_user['drs'])
+        intervals = [get_start_end_date(name) for name in files]
+
+        min_date = min(intervals)[0]
+        max_date = max(intervals)[1]
```

Comment:
I have pending to address this because I am surprised it does not make the tests fail.

>Or have the periods not been changed to datetimes there?

Periods are not changed to datetimes until the clipping, because I think that it's more reliable to work with the time coordinate in the cube, than the date that appears in the filename. However, since there is the check for missing years, I added the start and end year information in that step so that it appears in the `metadata.yml` file.